src/components/draw/DrawDirective.js

```diff
@@ -274,16 +274,18 @@ goog.require('ga_styles_service');
 
           // Set the layer to modify if exist, otherwise we use an empty
           // layer
-          layer = createDefaultLayer(map, useTemporaryLayer);
+          layer = layer || createDefaultLayer(map, useTemporaryLayer);
           if (!useTemporaryLayer) {
 
-            // If there is a layer loaded from public.admin.ch, we use it for
-            // modification.
-            map.getLayers().forEach(function(item) {
-              if (gaMapUtils.isStoredKmlLayer(item)) {
-                layer = item;
-              }
-            });
+            if (!layer.adminId) {
```

Comment:
In case another public kml file is added after the layer property  is already set, we don't want this new layer  replace the first one.